### PR TITLE
migrate hash-library to conan v2

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
This conan v2 migration PR was generated by automatoc script version 10.
It uses simple find and replace technics. Feel free to extend it.

### CODE REVIEW

**Review by uilianries — CHANGES_REQUESTED** [review]
Please, take into account the tools guidelines for your script

[docs.conan.io/en/latest/reference/conanfile/tools.html](https://docs.conan.io/en/latest/reference/conanfile/tools.html)

File: recipes/hash-library/all/conanfile.py [hunk 1]
```diff
@@ -53,7 +54,7 @@ def _configure_cmake(self):
 
     def build(self):
         for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+            tools.files.patch(self, **patch)
```
Comment (danimtb) [inline recipes/hash-library/all/conanfile.py:57]:
```suggestion
            patch(self, **patch)
```

File: recipes/hash-library/all/test_package/conanfile.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import os
 
-from conans import ConanFile, CMake, tools
+from conan import ConanFile, tools
+from conans import CMake
```
Comment (danimtb) [inline recipes/hash-library/all/test_package/conanfile.py:4]:
```suggestion
from conan import ConanFile
from conan.tools.build import cross_building
from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
```

File: recipes/hash-library/all/test_package/conanfile.py [hunk 3]
```diff
@@ -13,6 +14,6 @@ def build(self):
         cmake.build()
 
     def test(self):
-        if not tools.cross_building(self):
+        if not tools.build.cross_building(self):
```
Comment (danimtb) [inline recipes/hash-library/all/test_package/conanfile.py:17]:
```suggestion
        if not cross_building(self):
```

**Comment by prince-chrismc** [comment]
```
ERROR: hash-library/8.0 (test package): Error in test() method, line 25
	if can_run(self):
	NameError: name 'can_run' is not defined
```

File: recipes/hash-library/all/conanfile.py [hunk 4]
```diff
@@ -42,24 +43,18 @@ def configure(self):
...
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
-
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+       get(self, **self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
```
Comment (prince-chrismc) [inline recipes/hash-library/all/conanfile.py:46]: Linter

```suggestion
        get(self, **self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
```

File: recipes/hash-library/all/test_package/conanfile.py [hunk 5]
```diff
@@ -1,18 +1,27 @@
 import os
 
-from conans import ConanFile, CMake, tools
+from conan import ConanFile
+from conan.tools.build import cross_building
```
Comment (prince-chrismc) [inline recipes/hash-library/all/test_package/conanfile.py:4]:
```suggestion
from conan.tools.build import can_run
```

File: recipes/hash-library/all/conanfile.py [hunk 6]
```diff
@@ -13,17 +15,16 @@ class HashLibraryConan(ConanFile):
...
+        "shared": False,
         "fPIC": True
     }
-    generators = "cmake"
+    generators = "CMakeToolchain"
     exports_sources = ["CMakeLists.txt", "patches/*"]
 
-    _cmake = None
 
     @property
```
Comment (uilianries) [inline recipes/hash-library/all/conanfile.py:29]: please, remove both source_subfolder and build_subfolder, and use layout instead. Don't forget to update conandata.yml too, the patch there requires source_subfolder.

File: recipes/hash-library/all/conanfile.py [hunk 7]
```diff
@@ -1,7 +1,9 @@
-from conans import ConanFile, CMake, tools
-from conans.errors import ConanInvalidConfiguration
+from conan import ConanFile
+from conan.tools.cmake import CMake
+from conan.tools.files import apply_conandata_patches, get
+from conan.errors import ConanInvalidConfiguration
 
-required_conan_version = ">=1.33.0"
+required_conan_version = ">=1.52.0"
```
Comment (uilianries) [inline recipes/hash-library/all/conanfile.py:6]: please, add export_conandata_patches to justify 1.52.0 requirement

File: recipes/hash-library/all/test_package/CMakeLists.txt [hunk 8]
```diff
@@ -1,9 +1,6 @@
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.15)
```
Comment (uilianries) [inline recipes/hash-library/all/test_package/CMakeLists.txt:1]:
```suggestion
cmake_minimum_required(VERSION 3.1)
```
Keep 3.1 you can run `conan test` without `conan create`

File: recipes/hash-library/all/conanfile.py (hunk 6 above)
Comment (prince-chrismc) [reply recipes/hash-library/all/conanfile.py:29]: These are baked into the build script 🙈

**Review by prince-chrismc — APPROVED** [review]

File: recipes/hash-library/all/conanfile.py [hunk 9]
```diff
@@ -13,14 +15,14 @@ class HashLibraryConan(ConanFile):
...
         "fPIC": [True, False]
     }
     default_options = {
-        "shared": False, 
+        "shared": False,
         "fPIC": True
     }
-    generators = "cmake"
+    generators = "CMakeToolchain"
     exports_sources = ["CMakeLists.txt", "patches/*"]
```
[the remark this answers is not in the document]
Comment (uilianries) [reply recipes/hash-library/all/conanfile.py:26]: Up

**Pull request merged**